magento/magento2
# Fix webapi returns variables in JSON using snake case and I need camelCase

### Description (*)
Added objectKeyMap which accepts parameters from di.xml to modify data object keys
We can now modify the keys of any accessor on any class used in the rendering of the REST API, using the following (or similar) snippet in the `di.xml` of the `webapi_rest` directory. 
````
<type name="Magento\Framework\Reflection\DataObjectProcessor">
        <arguments>
            <argument name="objectKeyMap" xsi:type="array">
                <item name="Magento\Cms\Api\Data\PageInterface" xsi:type="array"> 
                    <item name="page_layout" xsi:type="string">PageLayout</item> 
                </item>
            </argument>
        </arguments>
    </type>
````

### Fixed Issues (if relevant)

1. Fixes magento/magento2#24681 : webapi returns variables in JSON using snake case and I need camelCase

### Manual testing scenarios (*)
1. Add a `di.xml` inside `webapi_rest` directory of any module and inject object key names to it, similar to the following
````
<type name="Magento\Framework\Reflection\DataObjectProcessor">
        <arguments>
            <argument name="objectKeyMap" xsi:type="array">
                <item name="Magento\Cms\Api\Data\PageInterface" xsi:type="array">
                    <item name="page_layout" xsi:type="string">PageLayout</item>
                </item>
            </argument>
        </arguments>
    </type>
````
2. Call the REST API and observe the new keys reflecting.

### Contribution checklist (*)
